docs/gcp-worker-guide.md

```diff
@@ -0,0 +1,307 @@
+- prefect.deployments.steps.set_working_directory:
+    directory: /opt/prefect/[WORKING_DIRECTORY]
+
+# the deployments section allows you to provide configuration for deploying flows
+deployments:
+- name: gcp-weather-deploy
+  version: null
+  tags: []
+  description: null
+  schedule: {}
+  flow_name: null
+  entrypoint: flows/weather_flow.py:fetch_weather
+  parameters: {lat: 14.5994,
+               lon: 28.6731}
+  work_pool:
+    name: my-cloud-run-pool
+    work_queue_name: default
+    job_variables:
+      image: '{{ build_image.image }}'
+```
+
+!!!Tip
+    After copying the example above, don't forget to replace `[WORKING_DIRECTORY]` with the name of the directory where your flow folder and prefect.yaml live.
+
+To get a better understanding of the different components of the `prefect.yaml` file above and what they do, feel free to read this next section. Otherwise, you can skip ahead to *Flow Deployment*.
+
+In the `build` section of the `prefect.yaml` file, there are two steps which are executed:
+
+1. `prefect.deployments.steps.run_shell_script` : runs a shell command which configures Docker to authenticate with GCR using your Google Cloud credentials.
+2. `prefect_docker.deployments.steps.build_docker_image` : builds a Docker image automatically which uses the name and tag chosen previously.
+
+!!!Warning
+    If you are using an M1 Mac, you'll want to ensure that you add `platform: linux/amd64` to your `build_docker_image` step to ensure that your docker image uses amd64 architecture. For example:
+
+    ```yaml 
+    - prefect_docker.deployments.steps.build_docker_image:
+    id: build_image
+    requires: prefect-docker>=0.3.1
+    image_name: gcr.io/prefect-project/gcp-weather-image
+    tag: latest
+    dockerfile: auto
+    platform: linux/amd64
+    ```
+
+
+The `push` section sends the Docker image to GCR, so that it can be easily accessed by the worker for flow run execution.
+
+In the `deployments` section of the `prefect.yaml` file above, you'll see that there is a deployment declaration named `gcp-weather-deploy`. Within the declaration, the entrypoint for the flow is specified along with some default parameters which will be passed to the flow at runtime. Last but not least, the name of the workpool that we created in step 2 of this guide is specified.
+
+####Flow Deployment
+Once you're happy with the specifications in the `prefect.yaml` file, run the following command in the terminal to deploy your flow:
+
+```bash
+prefect deploy --name gcp-weather-deploy
+```
+
+Once the flow is deployed to Prefect Cloud or your local Prefect Server, it's time to queue up a flow run!
+
+### Step 5. Flow Execution
+Find your deployment in the UI, and hit the *Quick Run* button.
+You have now successfully submitted a flow run to your Cloud Run worker!
+If you used the flow script provided in this guide, check the artifacts tab for the flow run once it completes.
+You'll have a nice little weather report waiting for you there.
```

Comment:
Love this - could add a "Hope your day is sunny!" or something to the end.

I suggest a quick recap and next steps sections at the end. Maybe direct to the Serverless Work Pools page in the Prefect docs as something they might be interested in.